Multitouch Support/Native/VoodooI2CNativeEngine.cpp

```diff
@@ -137,14 +139,32 @@ bool VoodooI2CNativeEngine::isForceClickEnabled() {
         lastForceClickPropertyUpdateTime = now_abs;
         return lastIsForceClickEnabled;
     }
-
+    
     // Blocking reading here takes about 10 microseconds
     OSDictionary* dict = OSDynamicCast(OSDictionary, this->getProperty("MultitouchPreferences", gIOServicePlane, kIORegistryIterateRecursively));
     if (!dict) {
         return lastIsForceClickEnabled;
     }
     if (OSCollectionIterator* i = OSCollectionIterator::withCollection(dict)) {
         while (OSSymbol* key = OSDynamicCast(OSSymbol, i->getNextObject())) {
+            // If tap to click (property "Clicking") is disabled,
+            // then we should forbid force clicking,
+            // because it makes impossible to perform a non-force click
+            if (key->isEqualTo("Clicking")) {
+                OSBoolean *boolValue;
+                OSNumber *numValue;
+                if ((boolValue = OSDynamicCast(OSBoolean, dict->getObject(key))) != NULL) {
```

Comment:
Please convert `NULL -> nullptr` (not just here)

Reply:
Done